### PR TITLE
Autotest: Added functions to arducopter autotest for motor failure testing

### PR DESCRIPTION
**Pull request by GuilhermeGSousa**
This PR firstly extends the changes made to sim_vehicle.py made in #5914 to include different parameter files by default on the arducopter autotest function. It also includes a function, using the changes from this same PR, I am using to test motor failure on hexacopters and quadcopters.

Apologies @peterbarker for closing the other PR (#6021) as I made a mistake rebasing to master and inadvertly closed it, the changes you mencioned were all added to this PR

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -924,6 +924,55 @@ def fly_mission(mavproxy, mav, height_accuracy=-1.0, target_altitude=None):
...
+                    output_message+="[oversaturated] "
+                if val < 1200:
+                    output_message+="[undersaturated] "
+                output_message+= "servo " + str(i+1) + " " + str(val) + "; "
+                output_message += '\n'
+
+        print(output_message)
+
+        print("Altitude drop of %f meters" % (start_altitude-hud_data.alt))
+        print("Yaw rate increased of %f rad/s" % (attitude_data.yawspeed-start_yaw_rate))
```
Comment (magicrub) [inline Tools/autotest/arducopter.py]: I like how this checks alt AND yaw. Good check! I see we're iterating over each motor, that's good. Are we also iterating this sort of check over a 3,4,5,6,8 bladed flying robot flavor? That is probably out of scope for this PR, just wondering if all frames are checked.

Comment (GuilhermeGSousa) [inline Tools/autotest/arducopter.py]: Yes it checks the 8 PWM signal outputs from the servos to which the motors connect. What would be really nice would be to test the signals of opposing motors (for paired motor frames) as when one fails that output oversaturates and the opposing one undersaturates to try to compensate. Unfortunately the servos of opposing motors aren't always consequent numbers (for the octas for example servos 3 and 6 are opposed) so I left it as it is as it's much cleaner this way.

**Comment by magicrub** [comment]
a more advanced feature would be to integrate the abs yaw/alt changes. That way we're not just looking to see if ti recovered from it after a given amount of time, but how much it freaks out during the failure process.

**Comment by magicrub** [comment]
Also, we may want to a few seconds of "normal" time between tests to ensure the aircraft stabalizes

**Comment by GuilhermeGSousa** [comment]
I'll add those changes, as for the yaw I opted by displaying the yaw rate because while testing this for an hexa it started to spin quite fast and I made it very difficult to read the yaw, it kept going from 0 to 359. It was clearer for me to analyse abs yaw change on APM from the .tlog file

**Comment by magicrub** [comment]
but that .tlog can't be used in the regression check, right?

**Comment by GuilhermeGSousa** [comment]
You're right I dont think it can. From the tests I made with hexa and quad those always crashed but I didn't try with other frames

**Comment by GuilhermeGSousa** [comment]
@magicrub I added the yaw and the changes integration, is this what you had in mind?

**Comment by peterbarker** [comment]
Closed in favour of https://github.com/ArduPilot/ardupilot/pull/8399

**Comment by peterbarker** [comment]
Closed in favour of https://github.com/ArduPilot/ardupilot/pull/8399

**Pull request closed without merging**